### PR TITLE
column: deprecate 'key_name' in favor of 'name'

### PR DESCRIPTION
**Pull request by jharley**
Using 'key_name' felt like we were leaking an API abstraction into the resource needlessly.

This PR also removes a surprising behavior in the column resource where an existing column would silently be used on create instead of erroring.

### CODE REVIEW

File: honeycombio/resource_column.go [hunk 1]
```diff
@@ -91,30 +103,29 @@ func resourceColumnCreate(ctx context.Context, d *schema.ResourceData, meta inte
 	client := meta.(*honeycombio.Client)
 
 	dataset := d.Get("dataset").(string)
-	column := readColumn(d)
-
-	existing, err := client.Columns.GetByKeyName(ctx, dataset, column.KeyName)
```
Comment (jharley) [inline honeycombio/resource_column.go:96]: I'm happy to break this out into a separate PR if we feel that's better

**Review by dstrelau — APPROVED** [review]
